### PR TITLE
Update monorepo

### PR DESCRIPTION
**Pull request by i1i1**
This pr adds some options and updates monorepo to support all the additions. Also it adds gemini 3a chainspec.

There are still some todo, for example support of running domain system (which as far as I remember should be disabled for gemini 3a).

### CODE REVIEW

**Review by achiurizo — APPROVED** [review]

File: src/farmer.rs [hunk 1]
```diff
@@ -22,6 +16,44 @@ use tokio::sync::{oneshot, watch, Mutex};
...
 
+/// Description of the cache
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct CacheDescription {
+    /// Path of the plot
+    pub directory: PathBuf,
+    /// Space which you want to pledge
+    pub space_pledged: ByteSize,
+}
```
Comment (ozgunozerk) [inline src/farmer.rs:27]: these comments are confusing imo. Path of the plot and space I want to pledge sounds like the exact same thing with `PlotDescription`. However, we both ned `Cache` and `Plot` in order to start farming I guess, so they should be different. 
These comments may highlight those differences maybe?

**Review by ozgunozerk — CHANGES_REQUESTED** [review]

File: src/farmer.rs [hunk 2]
```diff
@@ -22,6 +16,44 @@ use tokio::sync::{oneshot, watch, Mutex};
 
 use crate::{Node, PublicKey};
 
+/// Description of the cache
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct CacheDescription {
```
Comment (ozgunozerk) [inline src/farmer.rs:22]: can this derive Serialize/Deserialize?

**Review by ozgunozerk — CHANGES_REQUESTED** [review]

**Comment by ozgunozerk** [comment]
I tried to run with `gemini3a` spec in CLI using `ozgun-update-monorepo` branch. It stuck on `starting node...`, also with `--release`.
First time seeing this behavior... Are we sure on the SDK side that node will be able to start with gemini3a chain spec?

File: src/farmer.rs (hunk 2 above)
Comment (i1i1) [reply src/farmer.rs:22]: Sure

File: src/farmer.rs (hunk 1 above)
Comment (i1i1) [reply src/farmer.rs:27]: I mean, `CacheDescription` is description of cache, while `PlotDescription` is description of plot. I think their names speak clearly about them.

Comment (ozgunozerk) [reply src/farmer.rs:27]: `/// path of the plot` -> line 23

Comment (ozgunozerk) [reply src/farmer.rs:27]: So I'm pledging say 10GB space, denoted with PlotDescription,
and additional 1GB space, denoted with CacheDescription,
in total 11GB ?

I thought I'm only pledging space with my plot, not with DSN configuration. Maybe my understanding is wrong here

Comment (i1i1) [reply src/farmer.rs:27]: > So I'm pledging say 10GB space, denoted with PlotDescription,
>  and additional 1GB space, denoted with CacheDescription,
>  in total I'll earn rewards from 11GB ?

Nope, here 1Gb is just an overhead for DSN, you won't get any rewards now for it.

Comment (ozgunozerk) [reply src/farmer.rs:27]: Yes I thought so. Hence my first comment.
`CacheDescription` does not define the plot, nor the space pledged.
So far, `Pledge` in terms of `farming`, is always used for the space that you will earn rewards from. We were not pledging space to node, although node was occupying space back then.

The wording should not be `plot` and `pledge` imo. I find it very confusing

Comment (ozgunozerk) [reply src/farmer.rs:27]: And why should someone give more space to cache overhead? 
I mean, for node, it was archiving the whole history but was not contributing to rewards, so increasing the size for node would archive more, contributing to the decentralization without any profit to the farmer.

For increasing the `farm` size, the farmer would earn more rewards.

For increasing the cache size of dsn, why would anyone do that? Maybe comments can explain this as well (faster sync time, archiving more, etc.)

**Comment by ozgunozerk** [comment]
LGTM! 🎉

**Pull request merged**